### ISSUE TITLE
SQLite Events DB file (registry_events.db) not created

### ISSUE THREAD

**Issue by nickthegreek89** [Quiq/docker-registry-ui#39]
Hello,

I am using the latest image of your Docker Registry UI (Image ID '**bfa6e6145cf8**') and upon executing docker run command with (an empty) host directory mounted inside suggested container's directory '/**opt/data**' in order to have events logs, container is started successfully but I don't see the '**registry_events.db**' file under '/opt/data' container's directory, as I would normally expect. The same happens If I don't mount any host directory to container's '/opt/data' directory.

Thus, when making any action in Registry's UI (I have configured the Registry as needed in order to send event notifications to UI), I receive the following error in container's logs :

**events.event_listener - ERROR - Error creating a table: unable to open database file**

Am I missing something? Isn't the SQLite DB file supposed to be created out-of-the-box upon container's startup?

**PS** : During docker run command, I am mounting the config file that exists in your project inside the container.

**Comment by roman-vynar** [comment]
May be permissions? The file should be created out-of-the-box upon startup.

**Comment by nickthegreek89** [comment]
What kind of permissions? I am just running the latest version of the image plus mounting the needed configuration file ('**config.yml**'), as advised in documentation (first proposed 'docker run' command for spinning up the container).

How are permissions involved in this?

**Comment by roman-vynar** [comment]
Please provide more details, i.e. docker run command, the folder structure you mount etc.

**Comment by nickthegreek89** [comment]
As aforementioned, I am using the first proposed syntax for the docker run command. So, the command that I, eventually, run is the following one :

`docker run -d -p 5001:8000 --restart always -v /opt/docker-registry-ui/config.yml:/opt/config.yml:ro --name=registry-ui-test quiq/docker-registry-ui`

And the '**config.yml**' file has the content of the config file found on this project.

**Comment by roman-vynar** [comment]
Since you are not mounting anything to /opt/data, /opt may have been created with wrong permissions inside the container.
Try adding `-v /some_local_folder:/opt/data` to the docker command if you are going to use event listener functionality. Then the sqlite db file will be created in `/some_local_folder`

**Comment by nickthegreek89** [comment]
Ok, I will try it, but I don't think it should make any difference. This is in case I need data persistence. If not, mounting local directory is not needed and the db file should be created in any case (even with no data persistence).

In general, the db file's creation should happen if no folder is mounted from the host (so no permissions in the equation of this case), as far as I understand it.

Am I missing anything?

**Comment by roman-vynar** [comment]
Hmm, you do everything correctly, I really have no idea :-/

**Comment by nickthegreek89** [comment]
Yes, it seems strange to me also. That's why I opened the issue in case anyone else had the same issue or any idea on this.

Cannot figure out why this is really happening. :/ Have you tried the latest available image and it creates the db file normally?

**Comment by KaptinLin** [comment]
HI, i have this issue too.

```
registry-ui    | 2019-08-12 13:24:54 - events.event_listener - ERROR - Error creating a table: unable to open database file
```

**Comment by KaptinLin** [comment]
I'm using the latest docker-registry-ui

```
registry-ui:
    container_name: registry-ui
    image: quiq/docker-registry-ui:latest
    ports:
      - "8000"
    volumes:
      - $PWD/registry-ui/data:/opt/data
      - $PWD/registry-ui/config.yml:/opt/config.yml:ro
    networks:
      - frontend
    depends_on:
      - registry
```

**Comment by vivalenta** [comment]
fixed if insert full patch to db 

run `mkdir registry-ui/ && chmod -R 777 registry-ui/`
then `docker run -d -p 8000:8000 -v /home/vivalenta/tmp/config.yml:/opt/config.yml:ro -v /home/vivalenta/tmp/registry-ui:/data quiq/docker-registry-ui`

config.yml
`event_database_driver: sqlite3`
`event_database_location: /data/registry_events.db`

**Comment by roman-vynar** [comment]
WORKDIR is /opt in the Dockerfile.
So if you mount your data dir into container not into /opt/data you will need to re-define `event_database_location` in the config file as it is relative by default `data/registry_events.db `.

**Comment by gbouthenot** [comment]
The regsitry database folder should be chown 65534:65534
I think it should have been set in the entrypoint.sh but it is not the case.

**Comment by roman-vynar** [comment]
> The regsitry database folder should be chown 65534:65534

Yea, this is true as Dockerfile has `USER nobody`.

**Comment by roman-vynar** [comment]
Fixed the initial ownership in Dockerfile. So if you don't mount your data dir to /opt/data in the container, it will be properly owned by nobody. If you mount your own folder - you need to chown it to nobody (id 65534).

I mentioned this in the README as well. The fix to be released in 0.9.0.
Thanks!
